packages/openapi3-parser/lib/parser.js

```diff
@@ -126,14 +126,30 @@ function parse(source, context) {
     R.unless(isObjectOrAnnotation, createError(context.namespace, 'Source document is not an object')),
     R.unless(isAnnotation, parseOpenAPIObject(context)));
 
-  return R.chain(
+  const parseResult = R.chain(
     R.pipe(
       parseDocument,
       deduplicateUnsupportedAnnotations(context.namespace),
       context.options.generateSourceMap ? filterColumnLine : filterSourceMaps
     ),
     document
   );
+
+  if (!isAnnotation(parseResult.content[0])) {
+    const formatVersion = (/3\.\d+\.\d+/).exec(source)[0];
```

Comment:
This can have false positives because it will run the regex on the entire source document, if the input was something along the lines of the following:

```yaml
info:
  title: hello
  version: 3.2.0
openapi: 3.0.0
paths: {}
```

This would incorrectly get `paths` as it is assuming that the first `3.x.x` value is the version of OpenAPI:

```
$ pbpaste | npx fury - | jq .meta.links.content[0]
{
  "element": "link",
  "meta": {
    "title": {
      "element": "string",
      "content": "OpenAPI 3.2.0"
    }
  },
  "attributes": {
    "relation": {
      "element": "string",
      "content": "via"
    },
    "href": {
      "element": "string",
      "content": "https://spec.openapis.org/oas/v3.2.0"
    }
  }
}
```

Reply:
inserted a "look-behind" into the regex